arwen/host/asyncCall.go

```diff
@@ -117,12 +135,14 @@ func (host *vmHost) determineAsyncCallExecutionMode(asyncCallInfo *arwen.AsyncCa
 		return arwen.AsyncUnknown, err
 	}
 
-	sameShard := host.AreInSameShard(runtime.GetSCAddress(), asyncCallInfo.Destination)
+	actualDestination := host.determineDestinationForAsyncCall(asyncCallInfo.Destination, functionName, args)
```

Comment:
`determineDestinationForAsyncCall` parses esdt transfers but the check for `isESDTTransfer` is below this, so the `determineDestinationForAsyncCall` can be called without the action being an esdt transfer. Is it ok or should we add a check for `isESDTTransfer` inside the function?

Reply:
no need to add a check inside. as the parseESDTTransfer will have the check.